core-java-modules/core-java-reflection-2/src/test/java/com/baeldung/reflection/set/fields/SetFieldsUsingReflectionUnitTest.java

```diff
@@ -0,0 +1,184 @@
+        uidNumberField.setShort(person, uidNumber);
+        Assertions.assertEquals(5555, person.getUidNumber());
+
+        Field pinCodeField = person.getClass()
+            .getDeclaredField("pinCode");
+        pinCodeField.setAccessible(true);
+
+        int pinCode = 411057;
+        pinCodeField.setInt(person, pinCode);
+        Assertions.assertEquals(411057, person.getPinCode());
+
+        Field contactNumberField = person.getClass()
+            .getDeclaredField("contactNumber");
+        contactNumberField.setAccessible(true);
+
+        long contactNumber = 123456789L;
+        contactNumberField.setLong(person, contactNumber);
+        Assertions.assertEquals(123456789L, person.getContactNumber());
+
+    }
+
+    @Test
+    public void whenDoUnboxing_thenSuccess() throws Exception {
+        Person person = new Person();
+
+        Field pinCodeField = person.getClass()
+            .getDeclaredField("pinCode");
+        pinCodeField.setAccessible(true);
+
+        Integer pinCode = 411057;
+        pinCodeField.setInt(person, pinCode);
+        Assertions.assertEquals(411057, person.getPinCode());
+    }
+
+    @Test
+    public void whenDoNarrowing_thenSuccess() throws Exception {
+        Person person = new Person();
+
+        Field pinCodeField = person.getClass()
+            .getDeclaredField("pinCode");
+        pinCodeField.setAccessible(true);
+
+        short pinCode = 4110;
+        pinCodeField.setInt(person, pinCode);
+        Assertions.assertEquals(4110, person.getPinCode());
+    }
+
+    @Test
+    public void whenSetFloatingTypeFields_thenSuccess() throws Exception {
+        Person person = new Person();
+
+        Field heightField = person.getClass()
+            .getDeclaredField("height");
+        heightField.setAccessible(true);
+
+        float height = 6.1242f;
+        heightField.setFloat(person, height);
+        Assertions.assertEquals(6.1242f, person.getHeight());
+
+        Field weightField = person.getClass()
+            .getDeclaredField("weight");
+        weightField.setAccessible(true);
+
+        double weight = 75.2564;
+        weightField.setDouble(person, weight);
+        Assertions.assertEquals(75.2564, person.getWeight());
+    }
+
+    @Test
+    public void whenSetCharacterFields_thenSuccess() throws Exception {
+        Person person = new Person();
+
+        Field genderField = person.getClass()
+            .getDeclaredField("gender");
+        genderField.setAccessible(true);
+
+        char gender = 'M';
+        genderField.setChar(person, gender);
+        Assertions.assertEquals('M', person.getGender());
+    }
+
+    @Test
+    public void whenSetBooleanFields_thenSuccess() throws Exception {
+        Person person = new Person();
+
+        Field activeField = person.getClass()
+            .getDeclaredField("active");
+        activeField.setAccessible(true);
+
+        boolean active = true;
+        activeField.setBoolean(person, active);
```

Comment:
Avoid unnecessary variable declaration, change to just `activeField.setBoolean(person, true);`